File: JavaScript/test/permute_string_test.js
import { assert } from 'chai';
import permuteString from '../Strings/permute_string.js';

let string = 'ABC';

describe('permuteString', () => {
  it('Returns all permutations of "ABC" as an array', () => {
    assert.sameMembers(permuteString(string), [
      'ABC',
      'BAC',
      'CBA',
      'BCA',
      'CAB',
      'ACB',
    ]);
  });
});

File: Python/Strings/Permutations/tests/permute_string_test.py
import unittest
from ..permute_string import permute_string


class StringPermutationTest(unittest.TestCase):
    def test_sample_case(self):
        s = 'ABC'
        d = {}
        permutations = permute_string(s)

        self.assertEqual(len(permutations), 3 * 2 * 1)
        for i in range(len(permutations)):
            d[permutations[i]] = permutations[i]

        self.assertDictEqual(d,
                             {'ABC': 'ABC', 'BAC': 'BAC', 'CBA': 'CBA', 'BCA': 'BCA', 'CAB': 'CAB', 'ACB': 'ACB'})


if __name__ == '__main__':
    unittest.main()
